base/filesystem.jl

```diff
@@ -149,6 +149,26 @@ function read(f::File, ::Type{UInt8})
     return ret % UInt8
 end
 
+function read(f::File, ::Type{Char})
```

Comment:
Why is `function read(s::IO, ::Type{Char})` not sufficient here?

Reply:
This is annoying: `IO` doesn't support `position` and `seek`; `File` doesn't support `mark` and `reset`. That could be fixed instead, but this was the easiest way to get this working.